Update OTel GenAI conventions skill for standalone semconv-genai repo (#7519)
dotnet/extensions: .github/skills/update-otel-genai-conventions/SKILL.md

Analyze OpenTelemetry GenAI semantic-conventions changes — PRs, `CHANGELOG.md` snapshots, date ranges, or releases — primarily from [`open-telemetry/semantic-conventions-genai`](https://github.com/open-telemetry/semantic-conventions-genai), and produce compensating updates in `dotnet/extensions`. See the [Migration Note](#migration-note) below for the recent move from the core `semantic-conventions` repo and how the two repos relate.

## Migration Note

The OpenTelemetry GenAI semantic conventions recently moved from
[`open-telemetry/semantic-conventions`](https://github.com/open-telemetry/semantic-conventions)
(where they lived under the `area:gen-ai` label) to a dedicated repo:
[`open-telemetry/semantic-conventions-genai`](https://github.com/open-telemetry/semantic-conventions-genai).
The new repo also hosts `mcp`, `openai`, `anthropic`, `aws-bedrock`, and
`azure-ai-inference` conventions.

Implications for this skill:

- **Primary input source** is the new repo. The old repo is a **fallback**
  for catch-up audits, historical context, and any in-flight PR that
  started there before the migration.
- **No `area:gen-ai` label** exists in the new repo — every PR is in scope

>   by definition. The label still applies for old-repo catch-up work.

Reviewer: The repo does have `area:*` labels, just not `area:gen-ai`: it uses more granular ones like `area:mcp`, `area:inference`, `area:tools`, `area:embeddings`, and `area:streaming`. The "every PR is in scope" conclusion still holds, but it's worth noting these exist, since they're useful for triage. The later PR-references bullet ("No `area:` label filter is needed") also reads as if the repo has no area labels at all, so it could use the same tweak. Suggested wording here:

```suggestion
- **No `area:gen-ai` label** exists in the new repo, so every PR is in
  scope by definition (the `area:gen-ai` label still applies for old-repo
  catch-up work). The new repo does use more granular `area:*` labels (for
  example `area:mcp`, `area:inference`, `area:tools`, `area:embeddings`),
  which can help triage but are not required for scoping.
```

- **No releases yet** in the new repo.
  <!-- TODO: remove the "no releases yet" framing once semantic-conventions-genai ships its first release -->
  The `CHANGELOG.md` `Unreleased` section is the most reliable "what's new"
  view; pin a snapshot via commit SHA / ref for reproducible audits.
- **GenAI version is now independent** of core semconv. The schema URL
  `https://opentelemetry.io/schemas/gen-ai/X.Y.Z` carries the gen-ai
  version; core semconv tracks its own separate version line and is
  declared as a dependency in the new repo's `versions.env`.
- **Spec URL `https://opentelemetry.io/docs/specs/semconv/gen-ai/` is
  unchanged** — the published page still resolves and renders the gen-ai
  spec from the new repo, so the `<see href>` in dotnet/extensions source

>   files does not need to change.

Reviewer: `https://opentelemetry.io/docs/specs/semconv/gen-ai/` no longer renders the spec. It now returns a "Moved / no longer maintained" page that only links out to `semantic-conventions-genai`, so the `<see href>` in the `OpenTelemetry*` source files currently lands on that stub. Upstream also hasn't settled its own URL yet (the new repo's `README.md` `## Schema URL` is still `TODO`), so I'd keep the URL for now but soften this claim and leave a note to retarget the href once a canonical URL exists:

```suggestion
- **Spec URL `https://opentelemetry.io/docs/specs/semconv/gen-ai/`
  currently resolves to a "Moved" stub** that states the GenAI
  conventions have moved to the new repo and is no longer maintained; it
  no longer renders the spec content. The `<see href>` in dotnet/extensions
  source files still points at this URL, so leave it for now but revisit
  retargeting it once OpenTelemetry publishes a canonical URL for the new
  repo (the new repo's `README.md` `## Schema URL` section is currently
  `TODO`).
  <!-- TODO: once open-telemetry/semantic-conventions-genai publishes its
       canonical spec/schema URL, retarget the OpenTelemetry* doc-comment
       `<see href>` from the old /docs/specs/semconv/gen-ai/ stub to that URL. -->
```

| Upstream area (`model/<area>/`) | Maps to in dotnet/extensions |
|---|---|
| `gen-ai`, `gen-ai/agent` | `Microsoft.Extensions.AI` core (e.g. `OpenTelemetryChatClient`) |
| `mcp` | Currently no instrumentation; forward-looking — flag as a watch-list item if changes appear |
| `openai` | `Microsoft.Extensions.AI.OpenAI` |
| `anthropic` | Out of scope for `dotnet/extensions` today (no provider package). Implications land in [`anthropics/anthropic-sdk-csharp`](https://github.com/anthropics/anthropic-sdk-csharp) — apply this skill there per [Cross-repo applicability](#cross-repo-applicability). |
| `aws-bedrock` | Out of scope for `dotnet/extensions` today (no provider package). Implications land in the `BedrockRuntime` service library of [`aws/aws-sdk-net`](https://github.com/aws/aws-sdk-net) — apply this skill there per [Cross-repo applicability](#cross-repo-applicability). |
| `azure-ai-inference` | Corresponding provider package, if/when one exists in this repo; otherwise out of scope |

When classifying a change, identify its area from the path under
`model/<area>/` in the new repo (or from the doc page under

> `docs/<area>/`).

Reviewer: `model/` only contains `gen-ai`, `mcp`, `openai`, and `aws-bedrock` (plus `manifest.yaml`). There's no `model/anthropic/` or `model/azure-ai-inference/`; those two show up only as provider doc pages under `docs/gen-ai/` and as provider values in `model/gen-ai/registry.yaml`. The docs are also all under `docs/gen-ai/<provider>.md`, not `docs/<area>/`. So "identify the area from `model/<area>/` ... or `docs/<area>/`" breaks for `anthropic` and `azure-ai-inference`: you look for `model/anthropic/`, don't find it, and may treat the input as invalid. Suggested fix:

```suggestion
When classifying a change, identify its area as follows. `gen-ai`,
`mcp`, `openai`, and `aws-bedrock` each have a YAML registry under
`model/<area>/`, so use that path. `anthropic` and `azure-ai-inference`
do **not** have a `model/` registry today; they are documented only as
provider pages under `docs/gen-ai/<provider>.md`. All human-readable
docs live under `docs/gen-ai/` (for example `docs/gen-ai/openai.md`,
`docs/gen-ai/mcp.md`, `docs/gen-ai/anthropic.md`), not under
`docs/<area>/`.
```

The same correction applies to the other spots that imply all five areas have a `model/<area>/` registry or use a `docs/<area>/` layout: the "In-scope areas" table header `| Upstream area (`model/<area>/`) |`, the Mode 1 spec-fetch step, and the area lists in the Migration Note / Cross-repo sections.

1. Complete the **Existing dotnet/extensions PR Preflight** above. If a matching open PR exists, report it and stop.
2. **Determine the current implemented version**: Read the version reference from `OpenTelemetryChatClient.cs` doc comment to identify which convention version the codebase claims to implement
3. **Check for version drift**: Verify every file with a gen-ai semantic conventions version reference uses the same version. Use the search command from [references/file-inventory.md](references/file-inventory.md#version-references). If files reference different versions, flag that as a critical gap requiring investigation.

> 4. **Fetch the latest convention spec**: Read the current conventions from the [published spec](https://opentelemetry.io/docs/specs/semconv/gen-ai/) (rendered from the new repo) and cross-check against the source of truth in [`open-telemetry/semantic-conventions-genai`](https://github.com/open-telemetry/semantic-conventions-genai) — `docs/<area>/` for human-readable docs, `model/<area>/` for the YAML registry, and `schema-snapshot/` plus `README.md` for the current schema URL (e.g. `opentelemetry.io/schemas/gen-ai/X.Y.Z`). The new repo's gen-ai version is independent of core semconv, so verify both. Until releases exist in the new repo, use the latest `Unreleased` `CHANGELOG.md` entries or recently merged PRs as the "latest release notes" equivalent.

Reviewer: There's no `schema-snapshot/` directory in the repo (it 404s), and the schema URL `opentelemetry.io/schemas/gen-ai/X.Y.Z` isn't published yet: the `README.md` `## Schema URL` section is still `TODO`. So this step points at a path that doesn't exist and relies on a schema URL that isn't usable yet. This line also repeats the path and published-spec wording from the comments above, so the suggestion corrects those here too to keep the line consistent (published page is now a "Moved" stub, docs under `docs/gen-ai/`, and only `gen-ai`/`mcp`/`openai`/`aws-bedrock` have a `model/<area>/` registry):

```suggestion
4. **Fetch the latest convention spec**: Read the current conventions from the source of truth in [`open-telemetry/semantic-conventions-genai`](https://github.com/open-telemetry/semantic-conventions-genai): `docs/gen-ai/` for human-readable docs (for example `docs/gen-ai/gen-ai-spans.md`, `docs/gen-ai/openai.md`) and `model/<area>/` for the YAML registry (`gen-ai`, `mcp`, `openai`, `aws-bedrock`). Note the published page at `https://opentelemetry.io/docs/specs/semconv/gen-ai/` is currently a "Moved" stub and no longer renders the spec. There is no `schema-snapshot/` directory, and the schema URL (`opentelemetry.io/schemas/gen-ai/X.Y.Z`) is not published yet (the repo `README.md` `## Schema URL` section is `TODO`); until it is, derive the GenAI version from the `CHANGELOG.md` release header or `versions.env` rather than from the schema URL. The new repo's gen-ai version is independent of core semconv, so verify both. Until releases exist in the new repo, use the latest `Unreleased` `CHANGELOG.md` entries or recently merged PRs as the "latest release notes" equivalent.
<!-- TODO: once open-telemetry/semantic-conventions-genai publishes its schema URL (README ## Schema URL is currently TODO), switch the GenAI version source to that schema URL. -->
```
